lib/index.js

```diff
@@ -85,25 +95,96 @@ Vero.prototype.track = function(payload, fn) {
  * @api public
  */
 
-Vero.prototype.alias = request('/users/reidentify', 'put');
+Vero.prototype.alias = request('alias', '/users/reidentify', 'put');
+
+/**
+ * Add or Remove Tags.
+ *
+ * http://developers.getvero.com/?bash#tags
+ *
+ * @param {Facade} msg
+ * @param {Object} tags
+ * @param {Function} fn
+ * @api public
+ */
+
+Vero.prototype.addOrRemoveTags = function(msg, tags, fn) {
+  // Check to ensure tags object is properly formatted.
+  var errmsg;
+  if (Object.prototype.toString.call(tags) !== '[object Object]') {
+    errmsg = 'Tag option must be passed as an Object.';
+    return fn(new Error(errmsg));
+  }
+
+  if (!tags.action || tags.action !== 'add' && tags.action !== 'remove') {
+    errmsg = 'Tags action value must be either \'add\' or \'remove\'';
+    return fn(new Error(errmsg));
+  }
+
+  if (!tags.tags || !Array.isArray(tags.tags)) {
+    errmsg = 'Could not detect tag values. Tag values must be passed as array of strings.';
+    return fn(new Error(errmsg));
+  }
+
+  var path = '/users/tags/edit';
+  var payload = {};
+
+  payload.auth_token = this.settings.authToken;
+
+  // Add id to payload.
+  // Defaults to msg.userId() if id is not provided in options object.
+
+  if (tags.id) {
+    payload.id = tags.id;
+  } else {
+    payload.id = msg.userId();
+  }
+
+  // Either 'add' or 'remove'.
+  // See docs: http://developers.getvero.com/?bash#tags
+
+  payload[tags.action] = tags.tags;
+
+  this
+    .request('put', path)
+    .set('Accept', 'application/json')
+    .type('json')
+    .send(payload)
+    .end(fn);
+};
 
 /**
  * Request.
  *
+ * @param {String} integrationMethod
  * @param {String} path
- * @param {String} method
+ * @param {String} requestMethod
  * @return {Function}
  * @api private
  */
 
-function request(path, method){
-  method = method || 'post';
-  return function(payload, fn){
-    return this
-      .request(method, path)
+function request(integrationMethod, path, requestMethod) {
+  requestMethod = requestMethod || 'post';
+  return function(msg, fn) {
+    var payload = mapper[integrationMethod].call(this, msg);
+    var self = this;
+    return self
+      .request(requestMethod, path)
       .set('Accept', 'application/json')
       .type('json')
       .send(payload)
-      .end(this.handle(fn));
+      .then(function() {
+        var tags = checkForTags(msg);
+        if (tags) return self.addOrRemoveTags(msg, tags, fn);
+        return fn();
```

Comment:
here as well -- should return something but you're most likely going to have to change this up to use `Batch`